bindings/rascal/neighbourlist/structure_manager.py

```diff
@@ -57,19 +58,45 @@ def __iter__(self):
     def __getitem__(self, key):
         return self.managers[key]
 
-    def get_dense_feature_matrix(self, calculator):
+    def get_dense_feature_matrix(self, calculator, species=None):
         """
         Parameters
         -------
         calculator : Calculator (an object owning a _representation object)
 
+        species :  list of atomic number to use for building the dense feature
+        matrix computed with calculators of name Spherical*
+
         Returns
         -------
         represenation_matrix : ndarray
             returns the representation bound to the calculator as dense matrix.
         """
-        return self.managers.get_dense_feature_matrix(
+
+        if species is None:
+            X = self.managers.get_dense_feature_matrix(
                 calculator._representation)
+        else:
+            keys_list = calculator.get_keys(species)
+            X = self.managers.get_dense_feature_matrix(
+                calculator._representation, keys_list)
+
+        return X
+
+    def get_sparse_feature_matrix(self, calculator):
+        """
+        Parameters
+        -------
+        calculator : one of the representation calculators named Spherical*
+
+        Returns
+        -------
+        representation_matrix : dict of ndarray
+            returns the representation bound to the calculator as a dict of
+            dense matrix.
```

Comment:
This is not very clear to me. How about `returns a dictionary associating central atomic types to the corresponding feature matrices`?

Reply:
the keys are not central atomic types but tuples of atomic types sorted alphabetically. The atomic types do not need to only be the one of central atoms. 
Are you fine with  the new wording ?